From 92a49d9f04650358740bdc854022ddc918f9a8a9 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Thu, 10 Mar 2011 01:03:44 +0000
Subject: [PATCH] NET-365 FTPClient.listFiles() does not work properly, if
 remote server speaks German. Match non-space{3} instead of A-Za-z{3}

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/net/trunk@1080088 13f79535-47bb-0310-9956-ffa450edef68
---
 src/changes/changes.xml                                    | 7 ++++++-
 .../apache/commons/net/ftp/parser/UnixFTPEntryParser.java  | 6 ++++--
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 32a0fb35b..91de75d39 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -57,8 +57,13 @@ The <action> type attribute can be add,update,fix,remove.
 
     <body>
         <release version="3.0" date="TBA" description="TBA">
+            <action issue="NET-365" dev="sebb" type="fix">
+            FTPClient.listFiles() does not work properly, if remote server speaks German.
+            Match non-space{3} instead of A-Za-z{3}
+            </action>
             <action issue="NET-366" dev="sebb" type="fix">
-            FTPClientConfig: setServerLanguageCode and setShortMonthNames do not work
+            FTPClientConfig: setServerLanguageCode and setShortMonthNames do not work.
+            Ensure that config is passed to all parsers that can use it.
             </action>
             <action issue="NET-276" dev="sebb" type="fix">
             NNTPClient has problems with group listings for large groups.
diff --git a/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java b/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java
index e1b19190b..ba3e8a206 100644
--- a/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java
+++ b/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java
@@ -91,10 +91,12 @@ public class UnixFTPEntryParser extends ConfigurableFTPFileEntryParserImpl
         /*
          * numeric or standard format date:
          *   yyyy-mm-dd (expecting hh:mm to follow)
-         *   MMMM [d]d
+         *   MMM [d]d
          *   [d]d MMM
+         *   N.B. use non-space for MMM to allow for languages such as German which use
+         *   diacritics (e.g. umlaut) in some abbreviations.
         */
-        + "((?:\\d+[-/]\\d+[-/]\\d+)|(?:[a-zA-Z]{3}\\s+\\d{1,2})|(?:\\d{1,2}\\s+[a-zA-Z]{3}))\\s+"
+        + "((?:\\d+[-/]\\d+[-/]\\d+)|(?:\\S{3}\\s+\\d{1,2})|(?:\\d{1,2}\\s+\\S{3}))\\s+"
         /* 
            year (for non-recent standard format) - yyyy
            or time (for numeric or recent standard format) [h]h:mm  
